src/workflows-app/components/InputOutputModal.js

```diff
@@ -54,57 +54,63 @@ const InputOutputModal = ({ title, jsonData, onDismiss, sasToken }) => {
     },
     [
       div({ style: { margin: '1rem 0', display: 'flex', alignItems: 'center' } }, [
-        h(AutoSizer, { disableHeight: true }, [
-          ({ width }) =>
-            h(
-              FlexTable,
-              {
-                'aria-label': 'inputs outputs table',
-                height: tableHeight({ actualRows: dataArray.length, maxRows: 10.5 }), // The half-row here hints at there being extra rows if scrolled
-                width,
-                rowCount: dataArray.length,
-                noContentMessage: 'No Inputs',
-                columns: [
+        // When jsonData is null, we are expecting that an HTTP request is pending
+        // and so will show a loading spinner while it finishes fetching.
+        // When jsonData is undefined, that means that the HTTP request completed but fetched no data,
+        // so we show an empty table.
+        jsonData !== null
+          ? h(AutoSizer, { disableHeight: true }, [
```

Comment:
Consider flipping the branches here so that the entire ternary can be seen without scrolling.

```suggestion
        jsonData === null
          ? div([centeredSpinner()])
          : h(AutoSizer, { disableHeight: true }, [
```